include/libopencm3/stm32/common/dsi_common_f47.h

```diff
@@ -39,779 +39,1528 @@
-#define DSI_WPCR0_TDDL			(1 << 16)
-#define DSI_WPCR0_CDOFFDL		(1 << 14)
-#define DSI_WPCR0_FTXSMDL		(1 << 13)
-#define DSI_WPCR0_FTXSMCL		(1 << 12)
-#define DSI_WPCR0_HSIDL1		(1 << 11)
-#define DSI_WPCR0_HSIDL0		(1 << 10)
-#define DSI_WPCR0_HSICL			(1 << 9)
-#define DSI_WPCR0_SWDL1			(1 << 8)
-#define DSI_WPCR0_SWDL0			(1 << 7)
-#define DSI_WPCR0_SWCL			(1 << 6)
-#define DSI_WPCR0_UIX4_SHIFT		0
-#define DSI_WPCR0_UIX4_MASK		0x3f
-
-/**
- * DSI Wrapper PHY Configration Register 1
- */
-#define DSI_WPCR1			MMIO32(DSI_BASE + 0x41CU)
-#define DSI_WPCR1_LPRXFT_SHIFT		25
-#define DSI_WPCR1_LPRXFT_MASK		0x3
-#define DSI_WPCR1_FLPRXLPM		(1 << 22)
-#define DSI_WPCR1_HSTXSRCDL_SHIFT	18
-#define DSI_WPCR1_HSTXSRCDL_MASK	0x3
-#define DSI_WPCR1_HSTXSRCCL_SHIFT	16
-#define DSI_WPCR1_HSTXSRCCL_MASK	0x3
-#define DSI_WPCR1_SDDC			(1 << 12)
-#define DSI_WPCR1_LPSRCDL_SHIFT		8
-#define DSI_WPCR1_LPSRCDL_MASK		0x3
-#define DSI_WPCR1_HSTXDDL_SHIFT		2
-#define DSI_WPCR1_HSTXDDL_MASK		0x3
-#define DSI_WPCR1_HSTXDCL_SHIFT		0
-#define DSI_WPCR1_HSTXDCL_MASK		0x3
-
-/**
- * DSI Wrapper PHY Configuration Register 2
- */
-#define DSI_WPCR2			MMIO32(DSI_BASE + 0x420U)
-#define DSI_WPCR2_THSTRAIL_SHIFT	24
-#define DSI_WPCR2_THSTRAIL_MASK		0xff
-#define DSI_WPCR2_THSPREP_SHIFT		16
-#define DSI_WPCR2_THSPREP_MASK		0xff
-#define DSI_WPCR2_TCLKZERO_SHIFT	8
-#define DSI_WPCR2_TCLKZERO_MASK		0xff
-#define DSI_WPCR2_TCLKPREP_SHIF		0
-#define DSI_WPCR2_TCLKPREP_MASK		0xff
-
-/**
- * DSI Wrapper PHY Configuration Register 3
- */
-#define DSI_WPCR3			MMIO32(DSI_BASE + 0x424U)
-#define DSI_WPCR3_TLPXC_SHIFT		24
-#define DSI_WPCR3_TLPXC_MASK		0xff
-#define DSI_WPCR3_THSEXIT_SHIFT		16
-#define DSI_WPCR3_THSEXIT_MASK		0xff
-#define DSI_WPCR3_TLPXD_SHIFT		8
-#define DSI_WPCR3_TLPXD_MASK		0xff
-#define DSI_WPCR3_THSZERO_SHIFT		0
-#define DSI_WPCR3_THSZERO_MASK		0xff
-
-/**
- * DSI Wrapper PHY Configuration Register 4
- */
-#define DSI_WPCR4			MMIO32(DSI_BASE + 0x428U)
-#define DSI_WPCR4_TCLKPOST_SHIFT	0
-#define DSI_WPCR4_TCLKPOST_MASK		0xff
-
-/**
- * DSI Wrapper Regulator and PLL Control Register
- */
-#define DSI_WRPCR			MMIO32(DSI_BASE + 0x430U)
-#define DSI_WRPCR_REGEN			(1 << 24)
-#define DSI_WRPCR_ODF_SHIFT		16
-#define DSI_WRPCR_ODF_MASK		0x3
-#define DSI_WRPCR_ODF_DIV_1		0
-#define DSI_WRPCR_ODF_DIV_2		1
-#define DSI_WRPCR_ODF_DIV_4		2
-#define DSI_WRPCR_ODF_DIV_8		3
-#define DSI_WRPCR_IDF_SHIFT		11
-#define DSI_WRPCR_IDF_MASK		0xf
-#define DSI_WRPCR_IDF_DIV_1		1
-#define DSI_WRPCR_IDF_DIV_2		2
-#define DSI_WRPCR_IDF_DIV_3		3
-#define DSI_WRPCR_IDF_DIV_4		4
-#define DSI_WRPCR_IDF_DIV_5		5
-#define DSI_WRPCR_IDF_DIV_6		6
-#define DSI_WRPCR_IDF_DIV_7		7
-/* valid NDIV values 10 - 125 all other reserved */
-#define DSI_WRPCR_NDIV_SHIFT		2
-#define DSI_WRPCR_NDIV_MASK		0x7f
-#define DSI_WRPCR_PLLEN			(1 << 0)
+/** @defgroup dsi_registers DSI Registers
+ * @ingroup dsi_defines
+ * @brief DSI Host Control Registers
+@{*/
+
+/* --- DSI Host --- */
+/* DSI_VR:      DSI Host version */
```

Comment:
you've made it less lines, and i see you've tried to get the doxygen groupings, but you've dropped the second * in all your comments, so you've actually lost all the documentation that was existing, even if it was verbose.  

/** DSI Host version */ is fine, you can put the define in again if you feel like it, I find it a bit redundant personally, but the second * is critical.

Reply:
this style of commenting i took from the f4 rcc.h file.. I thought it is the standard way of commenting in libopencm3. It would be easy to change in bulk. Do you have a demo header whith comments somewhere?

Reply:
well, the f4 rcc defines don't show up in documentation either, so they're not a good example :)  With your changes, run "make doc" and look at what comes out.  (You don't get _any _ DSI docs after this change)

Reply:
The documentation should be fine now